include/coglu.h

```diff
@@ -15,20 +15,28 @@ unsigned int compile_shader(char *shader_path, int shader_type) {
         fprintf(stderr, "Could not open vertex shader file\n");
         exit(1);
     }
-    const char *shader_content = shader_source;
 
     // Compile shader
-    unsigned int shader;
-    shader = glCreateShader(shader_type);
+    const char *shader_content = shader_source;
+    unsigned int shader = glCreateShader(shader_type);
     glShaderSource(shader, 1, &shader_content, &length);
     glCompileShader(shader);
+
     int  success;
-    char info_log[512];
     glGetShaderiv(shader, GL_COMPILE_STATUS, &success);
     if (!success) {
-        glGetShaderInfoLog(shader, 512, NULL, info_log);
-        fprintf(stderr,"Shader compilation error in file %s: %s\n", shader_path, info_log);
+        int length;
```

Comment:
I would prefer `log_length` but that's a nitpick; it doesn't really matter but if you're gonna update this anyway might as well do that too :D